kore/src/Kore/Attribute/Sort/Constructors.hs

```diff
@@ -0,0 +1,37 @@
+{-|
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+
+-}
+
+module Kore.Attribute.Sort.Constructors
+    ( Constructors (..)
+    , ConstructorLike (..)
+    , Constructor (..)
+    ) where
+
+import Data.List.NonEmpty
+    ( NonEmpty
+    )
+
+import Kore.Internal.Symbol
+    ( Symbol
+    )
+import Kore.Sort
+    ( Sort
+    )
+
+data Constructor = Constructor
+    { name :: !Symbol
+    , sorts :: ![Sort]
+    }
+    deriving (Show, Eq)
+
+data ConstructorLike =
+    ConstructorLikeConstructor !Constructor
+  | ConstructorLikeInjection
+    deriving (Show, Eq)
+
+newtype Constructors =
+    Constructors { getConstructors :: Maybe (NonEmpty ConstructorLike) }
```

Comment:
Well, it was an attempt at making things more clear, i.e. "if we have a value for the constructors attribute, then the sort is constructor-based and the value is a list of constructors", as opposed to "if we have a value, then, if it's empty then the sort is not constructor-based, otherwise it's constructor-based and the value is a list of constructors". I guess I didn't make it more clear, should I just use `[ConstructorLike]`?